sachem/Controllers/RechercheInscriptionController.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Web;
+using System.Web.Mvc;
+using sachem.Models;
+using System.Net;
+using static sachem.Classes_Sachem.ValidationAcces;
+using System.Data.Entity;
+
+namespace sachem.Controllers
+{
+    public class RechercheInscriptionController : Controller
+    {
+        private readonly SACHEMEntities db = new SACHEMEntities();
+
+        const int BROUILLON = 1;
+        const int ACCEPTE = 3;
+        const int REFUSE = 5;
+
+        [ValidationAccesSuper]
+        public ActionResult Index()
+        {
+            var touteInscription = from inscription in db.Inscription
+                                   where inscription.id_Statut != BROUILLON
+                                   select inscription;
+
+            return View(Rechercher());
+        }
+
+        [ValidationAccesSuper]
+        public ActionResult Details(int? id)
+        {
+            if (id == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
+            }
+
+            Inscription inscriptionPersonne = db.Inscription.Find(id);
+            if(inscriptionPersonne == null)
+            {
+                return HttpNotFound();
+            }
+
+            List<Inscription> inscription = db.Inscription.Where(x => x.id_Pers == inscriptionPersonne.id_Pers).ToList();
+            RemplirDropList(inscription.First());
+            return View(inscription);
+        }
+
+        [HttpPost]
+        public ActionResult Edit(int id_Inscription, int id_Statut)
+        {
+            var inscription = db.Inscription.FirstOrDefault(x => x.id_Inscription == id_Inscription);
+            if(inscription != null && inscription.id_Sess == db.Session.Max(s => s.id_Sess))
+            {
+                inscription.id_Statut = id_Statut;
+                db.Entry(inscription).State = EntityState.Modified;
+                db.SaveChanges();
+
+                TempData["Success"] = "Inscription modifiée avec succès!";
+                return RedirectToAction("Details","RechercheInscription",new { id = id_Inscription });
+            }
+            TempData["Erreur"] = "Erreur lors de la modification de l'inscription. N'oubliez pas qu'il est impossible de modifier l'inscription des anciennes sessions.";
+            return RedirectToAction("Details", "RechercheInscription", new { id = id_Inscription });
+        }
+
+        [NonAction]
+        private void ListeSession(int Session = 0)
+        {
+            var lSessions = from session in db.Session select session;
+            var slSession = new List<SelectListItem>();
+            slSession.AddRange(new SelectList(lSessions, "id_Sess", "NomSession", Session));
+
+            ViewBag.Session = slSession;
+        }
+
+        [NonAction]
```

Comment:
l'attribut NonAction est inutile et redondant dans le cas d'une fonction privé car elle n'est pas accessible de l'extérieur.
Less code is better code